### DISCUSSION TITLE
iam-role cross-account flagging cognito-identities within own account

### DISCUSSION THREAD

**Discussion by khapp** [cloud-custodian/cloud-custodian#7420] [General]
### Describe the bug

When running an iam-role policy with the cross-account filter it finds a trust policy that contains a cognito-identity within its own account.

### What did you expect to happen?

I would expect the cognito-identity to be filtered out since it is within its own account, not cross-account.

### Cloud Provider

Amazon Web Services (AWS)

### Cloud Custodian version and dependency information

```shell
custodian version --debug

Please copy/paste the following info along with any bug reports:

Custodian:   0.9.13
Python:      3.8.10 (default, Jun  2 2021, 10:49:15)
             [GCC 9.4.0]
Platform:    posix.uname_result(sysname='Linux', nodename='ip-10-0-0-0', release='5.4.0-1030-aws', version='#31-Ubuntu SMP Fri Nov 13 11:40:37 UTC 2020', machine='x86_64')
Using venv:  True
Docker: False
Installed:

argcomplete==1.12.3
attrs==21.2.0
boto3==1.17.102
botocore==1.20.102
importlib-metadata==4.6.0
jmespath==0.10.0
jsonschema==3.2.0
pyrsistent==0.18.0
python-dateutil==2.8.1
pyyaml==5.4.1
s3transfer==0.4.2
setuptools==44.0.0
six==1.16.0
tabulate==0.8.9
typing-extensions==3.10.0.0
urllib3==1.26.6
zipp==3.4.1
```

### Policy

```shell
policies:

- name: iam-role-public-access
  resource: iam-role
  filters:
       - type: cross-account
```

### Relevant log/traceback output

```shell
Error received when policy is deployed: 
2021-09-22 15:18:08,486: custodian.iamaccess:WARNING no handler:handle_cognito_identity.amazonaws.com_aud op:StringEquals key:cognito-identity.amazo naws.com:aud values:('eu-west-1:855d1234-1234-1234-1234-123456789',)
```

### Extra information or context

Sample trust policy

{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Effect": "Allow",
      "Principal": {
        "Federated": "cognito-identity.amazonaws.com"
      },
      "Action": "sts:AssumeRoleWithWebIdentity",
      "Condition": {
        "StringEquals": {
          "cognito-identity.amazonaws.com:aud": "eu-west-1:855d1234-1234-1234-1234-123456789"
        },
        "ForAnyValue:StringLike": {
          "cognito-identity.amazonaws.com:amr": "authenticated"
        }
      }
    }
  ]
}

**Comment by kapilt** [comment]
its not clear what about this permission statement denotes that its within the same account. i would suggest setting an explicit allow on the cross account filter for the action or a separate value filter exception for the role if its a known. but in general federation can broadly be assumed to not be in the account.
